lib/events.js

```diff
@@ -664,6 +641,38 @@ function createIterResult(value, done) {
   return { value, done };
 }
 
+function addErrorHandlerIfEventEmitter(emitter, handler, flags) {
+  if (typeof emitter.on === 'function') {
+    eventTargetAgnosticAddListener(emitter, 'error', handler, flags);
+  }
+}
+
+function eventTargetAgnosticRemoveListener(emitter, name, listener, flags) {
+  if (typeof emitter.removeEventListener === 'function') {
```

Comment:
Nit: should we perhaps guard this with something like `[Symbol.toStringTag]` instead to make sure this doesn't break when custom functions (i.e. `removeEventListener`) are mixed into `EventEmitter` by users which would result in very confusing errors/listener disappears?

Reply:
I'm not sure I understand this part, are you concerned about implementations that implement an incorrect `removeEventListener` but a correct `.removeListener`? 

Would it help if we checked for the Emitter variants prior to the EventTarget ones?

Reply:
I was concerned with extensions/mixins of our `EventEmitter` that add methods like `addEventListener`/`removeEventListener` for their own use that would break this code since it would assume we are dealing with EventTarget when the method really is just a user-defined one that is most likely not the one we expect.

Basically my concern is that checking that `addEventListener`/`removeEventListener` are functions may not be sufficient to make sure we have an `EventTarget` and may break things as these names aren't that uncommon.

> Would it help if we checked for the Emitter variants prior to the EventTarget ones?

This will probably be better since EventEmitter has been present for a long time and has all kinds of different adaptations/variations is user code. Though this won't make the code future-proof for `EventTarget`. Hence I suggested using the `[Symbol.toStringTag]` to differentiate them (by checking `emitter[Symbol.toStringTag] === 'EventTarget'` and then defaulting to the `EventEmitter` case if that checked false)

Reply:
I will update the code to reflect that then.

This means we will prioritize EventEmitter over EventTarget here which makes sense.

Reply:
@lundibundi can you please take a look after the update?

I'd rather not rely on `toStringTag` since EventEmitter is the base class for a _lot_ of things and some of those things might override `toStringTag` (though we can recursively check the prototype chain or something if we want to)

Reply:
Yeah, `toStringTag` of `EventEmitter` is not an option but if we go with

> (by checking emitter[Symbol.toStringTag] === 'EventTarget' and then defaulting to the EventEmitter case if that checked false)

(so that `EventEmitter` is just the default and its `toStringTag` is never checked) that may work out better.

I'm not sure which of the options would be better since both just check the property anyway but at least `toStringTag` was designed with a similar purpose (to identify an object) in mind.

I'd be fine with both solutions but perhaps someone else may give their opinion.
/cc @mcollina @jasnell